wire-swift-generator/src/main/java/com/squareup/wire/swift/SwiftGenerator.kt

```diff
@@ -158,6 +158,14 @@ class SwiftGenerator private constructor(
       else -> null
     }
 
+  private val Field.defaultedValue: CodeBlock?
+    get() = default?.let {
+      return defaultFieldInitializer(type!!, it)
+    } ?: if (isMessage && !isRequiredParameter && !isCollection) {
+        val subType = schema.getType(type!!) as MessageType
+        if (subType!!.fields.all { !it.isRequiredParameter }) CodeBlock.of("%T()", subType.typeName) else null
+    } else null
```

Comment:
Would that read better?
```suggestion
  private val Field.defaultedValue: CodeBlock?
    get() {
      when (val default = default) {
        null -> {
          if (!isMessage || isRequiredParameter || isCollection) return null
          val subType = schema.getType(type!!) as MessageType
          if (subType.fields.any { it.isRequiredParameter }) return null
          return CodeBlock.of("%T()", subType.typeName)
        }

        else -> return defaultFieldInitializer(type!!, default)
      }
    }

```

Reply:
nice yes!